File: core/src/main/java/fr/planandchill/models/Customer.java
package fr.planandchill.models;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@Builder
public class Customer {
    private Integer id;
    private String firstName;
    private String lastName;
    private Boolean enabled;
    private String email;
    private String password;
    private String phoneNumber;
    private Boolean accountLocked = false;
    private LocalDateTime creationDate;
    private Boolean advertising;

    public Customer() {

    }
}

File: front/src/app/models/customer/customer.ts
export class Customer {
    id: number;
    firstName: string;
    lastName: string;
    phone: string;
    email: string;
    password: string;
    advertising: boolean;

    constructor(
        firstName: string = '',
        lastName: string = '',
        phone: string  = '',
        email: string  = '',
        password: string = '',
        advertising: boolean  = false
    ) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.phone = phone;
        this.email = email;
        this.password = password;
        this.advertising = advertising;
    }
}